docs/core/tools/custom-templates.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Create custom templates for dotnet new (.NET Core SDK 2.0 Preview 2) | Microsoft Docs
+description: 
+keywords: dotnet new, CLI, CLI command, .NET Core, template, templating
+author: guardrex
+ms.author: mairaw
+ms.date: 06/26/2017
+ms.topic: article
+ms.prod: .net-core
+ms.technology: dotnet-cli
+ms.devlang: dotnet
+ms.assetid: c28eb59d-4d18-4aba-bcd1-88bd887abc08
+---
+
+# Create custom templates for dotnet new
+
+[!INCLUDE [core-preview-warning](~/includes/core-preview-warning.md)]
+
+The [.NET Core SDK](https://www.microsoft.com/net/download/core) provides many useful templates, but you can create your own custom templates for any type of .NET project, such as an app, service, tool, or class library. You can even create a template that outputs one or more independent files, such as a configuration file.
+
+Templates work with any Microsoft framework, including .NET Core, .NET Framework, and Xamarin. Users install your custom template from a NuGet package on any NuGet feed, by referencing a NuGet *nupkg* file directly, or by specifying a file system directory that contains the template. The template engine offers features that allow you to replace values, include and exclude files and regions of files, and execute custom processing operations when your template is used.
```

Comment:
I'm not 100% feel'in this one because I really did mean to say that the template engine (as a tool) works with any "Microsoft framework" (as opposed to any other non-MS frameworks) and list (at least some of) them. However, I don't think this sentence helps much here. The main thrust of this paragraph has more to do with using them, and I think we can assume that everyone knows this is a MS technology. Let's try killing this sentence and see if that works better.

>